consult.el

```diff
@@ -576,10 +576,37 @@ Turn ARG into a list, and for each element either:
 - split it if it a string.
 - eval it if it is an expression."
   (mapcan (lambda (x)
-            (if (stringp x)
-                (split-string-and-unquote x)
-              (ensure-list (eval x 'lexical))))
-          (ensure-list arg)))
+	    (if (stringp x)
+		(split-string-and-unquote x)
+	      (ensure-list (eval x 'lexical))))
+	  (ensure-list arg)))
+
+(defun consult--build-args-with-paths (arg search-path-list)
+  "Return ARG and SEARCH-PATH-LIST joined as a flat list of split strings.
+
+Turn ARG into a list, and for each element either:
+- split it if it a string.
+- eval it if it is an expression.
+
+If SEARCH-PATH-LIST is non-nil and not equal to '(\".\"),
+splice its paths in at the end of the arg list."
+  (let ((args (copy-sequence (mapcan (lambda (x)
+				       (if (stringp x)
+					   (split-string-and-unquote x)
+					 (ensure-list (eval x 'lexical))))
+				     (ensure-list arg)))))
```

Comment:
The copy-sequence is unnecessary? Also you can reuse `consult--build-args` here.

Reply:
Next push will resolve these two.